niimpy/battery.py

```diff
@@ -13,37 +14,17 @@ def get_battery_data(battery, user=None, start = None, end = None):
     start: datetime, optional
     end: datetime, optional
     """
-    assert isinstance(battery, pd.core.frame.DataFrame), "data is not a pandas DataFrame"
-    
-    if(user!= None):
-        assert isinstance(user, str),"user not given in string format"
-        battery_data = battery[(battery['user']==user)]
-    else:
-        battery_data = battery
-        
-    if(start!=None):
-        start = pd.to_datetime(start)
-    else:
-        start = battery_data.iloc[0]['datetime']
-    if(end!= None):
-        end = pd.to_datetime(end)
-    else:
-        end = battery_data.iloc[len(battery_data)-1]['datetime']
-
-    battery_data = battery_data[(battery_data['datetime']>=start) & (battery_data['datetime']<=end)]
-
-    battery_data['battery_level'] = pd.to_numeric(battery_data['battery_level'])
-
-    #df['column'].fillna(pd.Timedelta(seconds=0))
-    #df.dropna()
-
-    battery_data = battery_data.drop_duplicates(subset=['datetime','user','device'],keep='last')
-    battery_data = battery_data.drop(['user','device','time','datetime'],axis=1)
+    battery = niimpy.filter_dataframe(battery, begin=start, end=end, user=user)
+    battery[batterylevel_column] = pd.to_numeric(battery[batterylevel_column])
 
-    return battery_data
+    battery = battery.drop_duplicates(subset=['datetime','user','device'],keep='last')
+    battery = battery.drop(['user','device','time','datetime'],axis=1)
```

Comment:
An example of depending on unrelated things.  Shouldn't need to drop things that may not be there, instead select what is important (or better yet, ignore others and let it be passed through.  It might be useful for someone?)